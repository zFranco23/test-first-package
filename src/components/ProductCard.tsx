import React from 'react';
import { createContext } from 'react';

import { useProduct } from '../hooks/useProduct';
import { ProductContextProps, Product, onChangeArgs, InitialValues} from '../interfaces/interfaces';

import styles from '../styles/styles.module.css';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;



export interface Props {
    product: Product;
    // children?: React.ReactElement | React.ReactElement[];
    children: (maxCount?: number) => JSX.Element;
    className?: string;
    style?: React.CSSProperties;
    onChange?: ( args: onChangeArgs ) => void;
    value?: number;
    initialValues?: InitialValues
}


export const ProductCard = (props: Props ) => {
    const { children, product, className, style, onChange, value, initialValues } = props;

    const { counter, increaseBy, maxCount } = useProduct({ onChange, product, value, initialValues });

    return (
        <Provider value={{
            counter,
            increaseBy,
            product,
            maxCount
        }}>
            <div 
                className={ `${ styles.productCard } ${ className }` }
                style={ style }
            >
                { children(initialValues?.maxCount) }
            </div>
        </Provider>
    )
}
