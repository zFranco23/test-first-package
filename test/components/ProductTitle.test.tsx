import renderer from 'react-test-renderer';
import React from 'react';

import {ProductTitle} from '../../src/components/ProductTitle';
import { product1 } from '../mock/products';
import { ProductCard } from '../../src/components/ProductCard';

describe('ProductTitle', () => {
  test('Should show correctly the component with the title', () => {
    const wrapper = renderer.create(<ProductTitle title="Hola"  className='custom-class'/>)


    expect(wrapper.toJSON()).toMatchSnapshot()    
  });

  test('Should show correctly the component with product name', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {() => <ProductTitle />}
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()    
  });
});
