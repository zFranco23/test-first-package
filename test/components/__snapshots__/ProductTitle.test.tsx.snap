// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Should show correctly the component with product name 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;

exports[`ProductTitle Should show correctly the component with the title 1`] = `
<span
  className="productDescription custom-class"
>
  Hola
</span>
`;
